### ISSUE TITLE
Stack and split error

### ISSUE THREAD

**Issue by dmlundeby** [hyperspy/hyperspy#2139]
The following code causes an IndexError:

```python
import hyperspy.api as hs
s = hs.datasets.example_signals.EDS_TEM_Spectrum()
m = s.create_model()
li = m.get_lines_intensity(xray_lines=['Fe_Ka'])
st = hs.stack(li)
li2 = st.split()
```

There is no error when retrieving intensities from `s` instead of `m`.
If there are multiple X-ray lines, there is no error.

Now, `m.get_lines_intensity` returns a scalar in the signal axis, while `m.get_lines_intensity` returns a scalar in the navigation axis. This might be the issue, but anyway I think stack and split should always be the reverse of each other.

**Comment by thomasaarholt** [comment]
You've actually identified two errors:

An even smaller example of the error you show is tested by stacking and splitting the signal itself:
```python
>>> import hyperspy.api as hs
>>> s = hs.datasets.example_signals.EDS_TEM_Spectrum()
>>> hs.stack([s]).split()
...
~\AppData\Local\Continuum\anaconda3\envs\py36\lib\site-packages\hyperspy\axes.py in _axes_getter(self, y)
    817         elif y.imag == 1:  # Navigation natural order
    818             #
--> 819             return self.navigation_axes[int(y.real)]
    820         elif y.imag == 2:  # Signal natural order
    821             return self.signal_axes[int(y.real)]

IndexError: tuple index out of range
```
This error probably comes from stacking only a single signal.

The other error (that .get_lines_intensity is inconsistent) is actually previously reported (#1418), and supposedly fixed (#1440)! (it is not fixed) @francisco-dlp want to comment on this? :)

**Comment by ericpre** [comment]
This is fixed in #2169.
